Pages/Pag81751.NSPPerformanceTest.al

```diff
@@ -152,119 +195,177 @@ page 81751 "NSP Performance Test"
         StartDateTime: DateTime;
         i: Integer;
+        Message: Text[250];
+        StartSqlStatements, StartSqlRowsRead : BigInteger;
     begin
+        DeleteEntries();
+        StartSqlStatements := SessionInformation.SqlStatementsExecuted;
+        StartSqlRowsRead := SessionInformation.SqlRowsRead;
         if not IsEntryTableReady then
             Error('You must Init Entry Table before you go.');
 
         StartDateTime := CurrentDateTime();
-        for i := 1 to NumberOfIterations do begin
-            EntryTable2.Init();
-            EntryTable2."Entry No." := 0;
-            EntryTable2.Description := 'test';
-            EntryTable2.Insert();
-        end;
-        Message('Auto Increment on Insert: %1 iterations took %2.', NumberOfIterations, CurrentDateTime() - StartDateTime);
+        for i := 1 to NumberOfIterations do
+            CreateEntry2();
+
+        Result.Log('TestAutoIncrementInsertRecord', CurrentDateTime() - StartDateTime, NumberOfIterations, comment, SessionInformation.SqlRowsRead - StartSqlRowsRead, SessionInformation.SqlStatementsExecuted - StartSqlStatements);
+        if ShowMessage then
+            Message('Auto Increment on Insert: %1 iterations took %2.', NumberOfIterations, CurrentDateTime() - StartDateTime);
+
+        Commit;
     end;
 
-    local procedure TestNumberIncrementFindLast()
+    local procedure TestNumberIncrementFindLast(ShowMessage: Boolean)
     var
         ListOfText: List of [Text];
         StartDateTime: DateTime;
         DummyEntryNo: Integer;
         i: Integer;
+        StartSqlStatements, StartSqlRowsRead : BigInteger;
     begin
+        DeleteEntries();
+        StartSqlStatements := SessionInformation.SqlStatementsExecuted;
+        StartSqlRowsRead := SessionInformation.SqlRowsRead;
         if not IsEntryTableReady then
             Error('You must Init Entry Table before you go.');
 
         StartDateTime := CurrentDateTime();
         for i := 1 to NumberOfIterations do begin
             DummyEntryNo := SimulateGetNextEntryNo(i - 1);
-            ListOfText.Add(Format(i));
+            //ListOfText.Add(Format(i));
+            CreateEntry(DummyEntryNo, true);
         end;
-        Message('Number increment: %1 iterations took %2.', NumberOfIterations, CurrentDateTime() - StartDateTime);
+        Result.Log('TestNumberIncrementFindLast+1', CurrentDateTime() - StartDateTime, NumberOfIterations, comment, SessionInformation.SqlRowsRead - StartSqlRowsRead, SessionInformation.SqlStatementsExecuted - StartSqlStatements);
+        if ShowMessage then
+            Message('Number increment: %1 iterations took %2.', NumberOfIterations, CurrentDateTime() - StartDateTime);
+        Commit;
     end;
 
     local procedure SimulateGetNextEntryNo(CurrentCounter: Integer): Integer
     var
         EntryTable: Record "NSP Entry Table";
     begin
-        EntryTable.SetRange("Entry No.", CurrentCounter);
+        //EntryTable.SetRange("Entry No.", CurrentCounter);
```

Comment:
@NAVFreak 
forbidden code

Reply:
As I said, I wanted to compare/simulate insertions that mimics that a lot of different do inserts at the same time.
That filter doesn't make any sense. Because the biggest drawback with doing findlast +1 is the time it takes to actually do the findlast and locking it that is the performance drainer. The filter combined with findlast could be switched to a get instead. Kind of the same thing but I didn't want to test get.